vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1706 @@
+                    int pin_id = sub_tile.sub_tile_to_tile_pin_indices[port.index];
+                    std::tuple<int, int, std::string> key{tile_type_id, it, std::string(port.name)};
+                    std::tie<float, int>(value, wire_id) = tile_type_site_num_pin_name_model[key];
+                    if (wire_to_node_.find(std::make_tuple(tile_id, wire_id)) == wire_to_node_.end()) {
+                        node_id = -1;
+                    } else {
+                        node_id = wire_to_node_[std::make_tuple(tile_id, wire_id)];
+                    }
+
+                    location loc = tile_to_loc_[tile_id];
+
+                    used_by_pin_.insert(std::make_tuple(node_id, loc));
+                    usefull_node_.insert(node_id);
+                    sink_source_loc_map_[tile_id][pin_id] = std::make_tuple(port.type == IN_PORT, value, node_id);
+                }
+                it = next_good_site(it + 1, tile);
+            }
+        }
+        /*
+         * Add constant ource
+         */
+        sink_source_loc_map_[-1].resize(2);
+        for (auto i : {0, 1}) {
+            location loc = tile_to_loc_[-1];
+            used_by_pin_.insert(std::make_tuple(i, loc));
+            usefull_node_.insert(i);
+            sink_source_loc_map_[-1][i] = std::make_tuple(false, 0, i);
+        }
+    }
+
+    void sweep(location loc,
+               e_rr_type type,
+               int& used_tracks,
+               std::unordered_map<int, std::list<int>>& sweeper,
+               std::list<int>& avaiable_tracks) {
+        for (size_t k = 0; k < virtual_chan_loc_map_[std::make_tuple(loc, type)].size(); ++k) {
+            auto track = virtual_chan_loc_map_[std::make_tuple(loc, type)][k];
+            int new_id;
+            if (!avaiable_tracks.empty()) {
+                new_id = avaiable_tracks.front();
+                avaiable_tracks.pop_front();
+            } else {
+                new_id = used_tracks++;
+            }
+            virtual_beg_to_real_[std::make_tuple(loc, type, k)] = new_id;
+            chan_loc_map_[std::make_tuple(loc, type)][new_id] = track;
+            int stop = type == CHANX ? std::get<3>(track).first : std::get<3>(track).second;
+            sweeper[stop].push_back(new_id);
+        }
+        int pos = type == CHANX ? loc.first : loc.second;
+        while (!sweeper[pos].empty()) {
+            avaiable_tracks.push_back(sweeper[pos].front());
+            sweeper[pos].pop_front();
+        }
+        virtual_chan_loc_map_[std::make_tuple(loc, type)].clear();
+    }
+
+    /*
+     * Create mapping from virtual to physical tracks.
+     * It should work in O(N*M + L), where n,m are device dims and l is number of used segments in CHANs
+     */
+    void virtual_to_real_() {
+        device_ctx_.chan_width.x_list.resize(grid_.height(), 0);
+        device_ctx_.chan_width.y_list.resize(grid_.width(), 0);
+
+        int min_x, min_y, max_x, max_y;
+        min_x = min_y = 0x7fffffff;
```

Comment:
You may use `std::numeric_limits<int>::max()`

Reply:
Fixed in next commit